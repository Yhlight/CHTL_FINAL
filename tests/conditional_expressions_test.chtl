// Test file for conditional property expressions

div {
    id: "box";
    style {
        width: 100px;
        height: 200px;
        opacity: 0.8;
    }
}

div {
    id: "test-box";
    style {
        // Simple conditional
        color: #box.width > 50px ? "red" : "blue"; // Should be red

        // Conditional with property reference in value
        height: #box.opacity > 0.5 ? #box.height : 100px; // Should be 200px

        // Conditional with arithmetic in branches
        width: #box.width == 100px ? (100px + 50px) : 50px; // Should be 150px

        // Conditional with logical operators
        border: #box.width > 50px && #box.height > 150px ? "1px solid green" : "none"; // Should be green border

        // False condition
        background-color: #box.width > 200px || #box.opacity > 1 ? "purple" : "yellow"; // Should be yellow
    }
}
