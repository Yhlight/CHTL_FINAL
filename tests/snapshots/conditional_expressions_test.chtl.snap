<div id="box" style="height: 200px; opacity: 0.8; width: 100px; " />
<div id="test-box" style="background-color: yellow; border: 1px solid green; color: red; height: 200px; width: 150px; " />
